### DISCUSSION TITLE
Scale Resources Based On Town/Nation Population

### DISCUSSION THREAD

**Discussion by EcoSMP** [TownyAdvanced/TownyResources#98] [Ideas]
At the moment our players are incentivized to all build individual towns instead of building towns together because they want to get more resources. We would love to further incentivize players to build towns together by scaling resources based on how many residents they have in their town/nation. Thanks for your consideration!

**Comment by LlmDl** [comment]
The way it works now is in order to make a 2nd, 3rd survey they have to meet a number of thresholds including their town_level (which is based on population.

You might be thinking about having 1st survey limited to weaker resources and it progressing forwards like is talked about in this ticket: https://github.com/TownyAdvanced/TownyResources/issues/72.

**Reply by EcoSMP** [reply]
Yes we could make that a requirement but as of right now the only requirement is how much land the town has claimed to allow players who prefer to build solo to do that. We would still like to let those types of players have success while further incentivizing building towns together by scaling the resources. Making a resident limit to do more surveys only incentivizes it by punishing players who don't build together, not by rewarding those who do. Does that make sense?

**Reply by LlmDl** [reply]
If they have not got the town_level permitting their 2nd level resource their town will not produce it. 

A town with lots of townblocks (usually also gotten via town_level or population) can survey many tiers but will not produce the resource unless they have the required town_level.

If the goal is to stop one or two player towns, you want the 1st tier resources to not be strategic/good.

**Reply by EcoSMP** [reply]
Our goal is not to stop one or two player towns. Many players really enjoy building solo or in small groups so we want to allow them to have success while incentivizing players to create towns together. We do not want to punish players who do want to build solo/smaller towns. The idea is to reward those who choose to build together without punishing those who don't want to.

We would be punishing solo or small towns pretty severely by making the level 1-2 resources bad. If anything, we would have to make the level 1-2 resources better so that the solo/small towns can still earn at least a little bit since they wouldn't be able to do more surveys since they don't have enough residents.

The only way we can really see to do this would be to add a simple multiplier option in the config similar to level configuration of the regular towny plugin how there's things like townBlockLimit, upkeepModifier, etc. We could potentially add a townresourcemodifier section that increases resource production at all levels if the town has more residents. 

I hope this makes sense. We want to encourage players to build together without punishing them for choosing to build solo/small towns.

**Reply by LlmDl** [reply]
I think its making a bit more sense, the trick will be making it fit in with or alongside the existing system.

**Reply by EcoSMP** [reply]
Well we greatly appreciate you hearing us out and considering this. Let us know if there's anything we can do to help!

**Comment by EcoSMP** [comment]
Hey, just wanted to check again if there's anything we can do to help with this? This change would be greatly appreciated throughout our community so if there's anything we can do to help please let us know!

**Reply by LlmDl** [reply]
This might be something I can put time into once I have the following off my plate:

1. 0.98.4.0 release
2. Alliance wartype added to EventWar
3. UUID db in its finished stages.

Also going through a bit of a thing with my IRL work-work which is falling apart and I have to figure out a new income source to feed/house my family.

**Reply by EcoSMP** [reply]
Thats understandable. Make sure you take care of you and your family before anything else.

**Comment by LlmDl** [comment]
This was added using a new town and nation_level stat.
